# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: docshell/base/CanonicalBrowsingContext.h
# Commit: ab54092b8afc
# Full Hash: ab54092b8afc8e56dd3f8162344a4ce38ec45999
# Author: Niklas Goegge <ngogge@mozilla.com>
# Date: 2021-04-29 09:26:05
# Regressor Bug: 1708150
# File Overlap Count: 1
# Description:
#   Bug 1708150 - Add user activation flag to reload, goBack and goForward r=ckerschb,Gijs,smaug
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D110245
# ==============================================================================

diff -r a9a52f01de13 -r ab54092b8afc docshell/base/CanonicalBrowsingContext.h
--- a/docshell/base/CanonicalBrowsingContext.h	Wed Apr 28 11:09:38 2021 +0000
+++ b/docshell/base/CanonicalBrowsingContext.h	Wed Apr 28 11:26:49 2021 +0000
@@ -166,7 +166,7 @@
   void RemoveFromSessionHistory(const nsID& aChangeID);
 
   void HistoryGo(int32_t aIndex, uint64_t aHistoryEpoch,
-                 bool aRequireUserInteraction,
+                 bool aRequireUserInteraction, bool aUserActivation,
                  Maybe<ContentParentId> aContentId,
                  std::function<void(int32_t&&)>&& aResolver);
 
@@ -203,11 +203,11 @@
                ErrorResult& aError);
 
   void GoBack(const Optional<int32_t>& aCancelContentJSEpoch,
-              bool aRequireUserInteraction);
+              bool aRequireUserInteraction, bool aUserActivation);
   void GoForward(const Optional<int32_t>& aCancelContentJSEpoch,
-                 bool aRequireUserInteraction);
-  void GoToIndex(int32_t aIndex,
-                 const Optional<int32_t>& aCancelContentJSEpoch);
+                 bool aRequireUserInteraction, bool aUserActivation);
+  void GoToIndex(int32_t aIndex, const Optional<int32_t>& aCancelContentJSEpoch,
+                 bool aUserActivation);
   void Reload(uint32_t aReloadFlags);
   void Stop(uint32_t aStopFlags);
 